tests/test_db.py

```diff
@@ -0,0 +1,87 @@
+from asyncio import AbstractEventLoop
+from typing import List
+
+import pytest
+import testing.postgresql
+from asyncpg import Record
+
+from parking.backend.db.dbaccess import DbAccess
+from parking.shared.location import Location
+from parking.shared.rest_models import ParkingLot
+
+
+@pytest.mark.asyncio
+async def test_empty_parking_lots(event_loop: AbstractEventLoop):
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+        async with db.pool.acquire() as conn:
+                lst = await conn.fetch('SELECT * from ParkingLots;')
+                assert len(lst) == 0
+
+
+@pytest.mark.asyncio
+async def test_insert_parking_lot(event_loop):
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        park_id: id = await db.insert_parking_lot(parking_lot)
+        assert park_id == 1
+
+        async with db.pool.acquire() as conn:
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+
+
+@pytest.mark.asyncio
+async def test_delete_parking_lot(event_loop):
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        await db.insert_parking_lot(parking_lot)
+        async with db.pool.acquire() as conn:
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 1
+            park_id: int = await db.delete_parking_lot(1)
+            assert park_id == 1
+            lst: List[Record] = await conn.fetch('SELECT * from ParkingLots;')
+            assert len(lst) == 0
+
+
+@pytest.mark.asyncio
+async def test_update_parking_lot_price_and_availability(event_loop):
+    new_availability = 10
+    new_price = 5
+
+    with testing.postgresql.Postgresql() as postgresql:
+        db = await DbAccess.create(postgresql.url(), event_loop, init_tables=True)
+
+        parking_lot = ParkingLot(100, 'test_name', 0.0, Location(0.0, 1.0))
+        park_id: id = await db.insert_parking_lot(parking_lot)
+        park_id: id = await db.update_parking_lot_availability(park_id, new_availability)
+        assert park_id == 1
+        park_id: id = await db.update_parking_lot_price(park_id, new_price)
```

Comment:
These type hints are wrong, shouldn't they be ```Optional[int]```?

Reply:
True.